### PR TITLE
Jvenstad/remove old job status

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review.

This is WIP, and there is quite some cleanup to do. Nevertheless, I think this all works now, i.e., the new deployment spec syntax, with nested parallel and serial blocks of steps, and with declared production tests. 

Some APIs need to remain until replacements have been available for a while, and some cleanup will have to wait for this as well. 

@bratseth FYI.

### CODE REVIEW

**Comment by bratseth** [comment]
Great! If/once we have enough testing to know this works we need to put out a post letting people know about these new capabilities. There are many hacks out there to compensate for not having multiple instances, and more people should have prod tests.

**Comment by jonmv** [comment]
I'm pretty confident it works, with the unit scenario tests we have, but documentation is not done. I would also like to have this verified in the CD system, of course. 
In any case, we should ask anyone with production tests to declare these with the new syntax, so we may stop running them as part of the deployment job.

**Comment by jonmv** [comment]
Ok, so there are some kinks to straighten out ... Expect some more commits :)

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/ApplicationPackageBuilder.java [hunk 1]
```diff
@@ -99,6 +99,13 @@ public ApplicationPackageBuilder region(String regionName) {
         return this;
     }
 
+    public ApplicationPackageBuilder test(String regionName) {
+        environmentBody.append("      <test>");
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/ApplicationPackageBuilder.java]: Have we decided on this tag name? `<test>` already means something else.

**Review by mpolden — APPROVED** [review]

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/ApplicationPackageBuilder.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/ApplicationPackageBuilder.java]: Yes, it was through a review.

**Pull request merged**